goth_tests/helpers/payment.py

```diff
@@ -120,16 +120,46 @@ class AllocationCtx:
     _id: Optional[str] = None
 
     async def __aenter__(self):
-        allocation = ya_payment.Allocation(
-            allocation_id="",
-            total_amount=str(self.amount),
-            spent_amount=0,
-            remaining_amount=0,
-            make_deposit=True,
-            timestamp=datetime.now(timezone.utc),
-            payment_platform=self.requestor.payment_config.platform_string,
-        )
-        allocation = await self.requestor.api.payment.create_allocation(allocation)
+        allocation = None
+        if global_deposits:
+            for global_deposit in global_deposits:
+                try:
+                    logger.info("Creating allocation for deposit {}".format(global_deposit))
+
+                    allocation_arg = ya_payment.Allocation(
+                        allocation_id="",
+                        total_amount=str(self.amount),
+                        spent_amount=0,
+                        remaining_amount=0,
+                        make_deposit=True,
+                        timestamp=datetime.now(timezone.utc),
+                        timeout=datetime.now(timezone.utc) + timedelta(minutes=30),
+                        payment_platform=self.requestor.payment_config.platform_string,
+                        deposit=global_deposit,
+                    )
+                    allocation = await self.requestor.api.payment.create_allocation(allocation_arg)
+                except Exception as ex:
+                    logger.warning("Failed to create allocation for deposit {} - {}".format(global_deposit["id"], ex))
+                    continue
```

Comment:
Why iterate over all deposits if you only care about the allocation for the last one?